### PR TITLE
Disable soft delete for Directions

### PR DESCRIPTION
**Pull request by alrom**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.DataAccess/Models/Direction.cs [hunk 1]
```diff
@@ -17,5 +17,7 @@ public class Direction : IKeyedEntity<long>
     [MaxLength(500)]
     public string Description { get; set; } = string.Empty;
 
+    public bool IsDeleted { get; set; }
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.DataAccess/Models/Direction.cs:20]: (Design considerations) As I can see it's common pattern to have `IsDeleted` property as a flag. Also there are lot of code checking it for this and other similar DTOs. Isn't it a reason to separate and use `ISoftDeleted` (or suggest your name) interface?

File: OutOfSchool/OutOfSchool.DataAccess/OutOfSchoolDbContext.SoftDelete.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ private void ApplySoftDelete(ModelBuilder builder)
             .ApplySoftDelete<CompanyInformation>()
             .ApplySoftDelete<CompanyInformationItem>()
             .ApplySoftDelete<DateTimeRange>()
-            .ApplySoftDelete<Direction>()
+            //.ApplySoftDelete<Direction>()
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.DataAccess/OutOfSchoolDbContext.SoftDelete.cs:61]: I'm not insisting, but maybe it's right time to remove this commented out code?

Comment (alrom) [reply OutOfSchool/OutOfSchool.DataAccess/OutOfSchoolDbContext.SoftDelete.cs:61]: We planned to clear it after finishing changing all entities. Commented lines show ours progress )))

Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.DataAccess/OutOfSchoolDbContext.SoftDelete.cs:61]: Code as a backlog? I'll steal this idea but would never ever use it )
And thanks for explanation

File: OutOfSchool/OutOfSchool.DataAccess/Models/Direction.cs (hunk 1 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.DataAccess/Models/Direction.cs:20]: Ok. We can use it in the EntityRepositoryBase to add predicate for queries. If you have another view how we can use this interface, please, write it. Thanks.

Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.DataAccess/Models/Direction.cs:20]: Sounds promising, Let's start with it and them see if it is beneficial

**Review by VadymLevkovskyi — APPROVED** [review]

**Pull request merged**